aea/mail/base.py

```diff
@@ -258,7 +298,9 @@ def encode(self, envelope: "Envelope") -> bytes:
         envelope_pb = base_pb2.Envelope()
         envelope_pb.to = envelope.to
         envelope_pb.sender = envelope.sender
-        envelope_pb.protocol_id = str(envelope.protocol_id)
+        envelope_pb.protocol_id = str(
+            self._get_protocol_specification_id_by_the_protocol_id(envelope.protocol_id)
+        )
```

Comment:
the translation should not happen here; it must happen in `aea.py` or at least be called from there